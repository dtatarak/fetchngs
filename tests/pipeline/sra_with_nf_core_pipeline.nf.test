nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    options "-profile test,docker"
    tag "pipeline"

    test("SRA NF-CORE/RNASEQ Samplesheet") {

        when {
            params {
                outdir           = "$outputDir"
                nf_core_pipeline = "rnaseq"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions") },
                { assert snapshot(path("$outputDir/fastq/").list()).match("fastq") },
                { assert snapshot(path("$outputDir/metadata/").list()).match("rnaseq_metadata") },
                { assert snapshot(path("$outputDir/samplesheet/id_mappings.csv"),
                                path("$outputDir/samplesheet/multiqc_config.yml")).match("rnaseq_samplesheet") },
                { assert new File("$outputDir/samplesheet/samplesheet.csv").exists() }
            )
        }
    }

    test("SRA NF-CORE/RNASEQ Samplesheet With Specific ENA metadata") {

        when {
            params {
                outdir                = "$outputDir"
                nf_core_pipeline      = "rnaseq"
                ena_metadata_fields   = "run_accession,experiment_accession,library_layout,fastq_ftp,fastq_md5"
                sample_mapping_fields = "run_accession,library_layout"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions") },
                { assert snapshot(path("$outputDir/fastq/").list()).match("fastq") },
                { assert snapshot(path("$outputDir/metadata/").list()).match("rnaseq_ena_metadata") },
                { assert snapshot(path("$outputDir/samplesheet/id_mappings.csv"),
                                path("$outputDir/samplesheet/multiqc_config.yml")).match("rnaseq_ena_samplesheet") },
                { assert new File("$outputDir/samplesheet/samplesheet.csv").exists() }
            )
        }
    }

    test("SRA NF-CORE/ATACSEQ Samplesheet") {

        when {
            params {
                outdir           = "$outputDir"
                nf_core_pipeline = "atacseq"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions") },
                { assert snapshot(path("$outputDir/fastq/").list()).match("fastq") },
                { assert snapshot(path("$outputDir/metadata/").list()).match("atacseq_metadata") },
                { assert snapshot(path("$outputDir/samplesheet/id_mappings.csv"),
                                path("$outputDir/samplesheet/multiqc_config.yml")).match("atacseq_samplesheet") },
                { assert new File("$outputDir/samplesheet/samplesheet.csv").exists() }
            )
        }
    }

    test("SRA NF-CORE/VIRALRECON Samplesheet") {

        when {
            params {
                outdir           = "$outputDir"
                nf_core_pipeline = "viralrecon"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions") },
                { assert snapshot(path("$outputDir/fastq/").list()).match("fastq") },
                { assert snapshot(path("$outputDir/metadata/").list()).match("viralrecon_metadata") },
                { assert snapshot(path("$outputDir/samplesheet/id_mappings.csv"),
                                path("$outputDir/samplesheet/multiqc_config.yml")).match("viralrecon_samplesheet") },
                { assert new File("$outputDir/samplesheet/samplesheet.csv").exists() }
            )
        }
    }

    test("SRA NF-CORE/TAXPROFILER Samplesheet") {

        when {
            params {
                outdir           = "$outputDir"
                nf_core_pipeline = "taxprofiler"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions") },
                { assert snapshot(path("$outputDir/fastq/").list()).match("fastq") },
                { assert snapshot(path("$outputDir/metadata/").list()).match("taxprofiler_metadata") },
                { assert snapshot(path("$outputDir/samplesheet/id_mappings.csv"),
                                path("$outputDir/samplesheet/multiqc_config.yml")).match("taxprofiler_samplesheet") },
                { assert new File("$outputDir/samplesheet/samplesheet.csv").exists() }
            )
        }
    }

}
