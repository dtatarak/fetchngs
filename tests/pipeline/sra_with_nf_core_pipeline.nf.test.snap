{
    "taxprofiler_metadata": {
        "content": [
            "DRR026872.runinfo_ftp.tsv:md5,c5ab92f6f50717b8163580b642b37aec",
            "DRR028935.runinfo_ftp.tsv:md5,9db5d6cba5ce2cccd9b2cc3faaa092e2",
            "ERR1109373.runinfo_ftp.tsv:md5,d1717cb1638950b7af22ec1b976616a4",
            "ERR1160846.runinfo_ftp.tsv:md5,5611622d583e6bc6d45d8f934696ff1e",
            "SRR12848126.runinfo_ftp.tsv:md5,9563f91d4cc6aa58d2ed5fba2eb30baf",
            "SRR13191702.runinfo_ftp.tsv:md5,94378c448c044b3e20e5c54e442ab62e",
            "SRR14593545.runinfo_ftp.tsv:md5,302271a4a1d99865362d0228e6439580",
            "SRR14709033.runinfo_ftp.tsv:md5,e059c4a0dfdddd70c1998a1ea9c1e65f",
            "SRR9984183.runinfo_ftp.tsv:md5,1e299191d23e0f3bd0aabf98c9d8133e"
        ],
        "timestamp": "2023-05-09T14:11:58+0000"
    },
    "software_versions": {
        "content": [
            "software_versions.yml:md5,c8ccda032b4de8e103eace767442fcf4"
        ],
        "timestamp": "2023-05-09T14:11:58+0000"
    },
    "viralrecon_metadata": {
        "content": [
            "DRR026872.runinfo_ftp.tsv:md5,c5ab92f6f50717b8163580b642b37aec",
            "DRR028935.runinfo_ftp.tsv:md5,9db5d6cba5ce2cccd9b2cc3faaa092e2",
            "ERR1109373.runinfo_ftp.tsv:md5,d1717cb1638950b7af22ec1b976616a4",
            "ERR1160846.runinfo_ftp.tsv:md5,5611622d583e6bc6d45d8f934696ff1e",
            "SRR12848126.runinfo_ftp.tsv:md5,9563f91d4cc6aa58d2ed5fba2eb30baf",
            "SRR13191702.runinfo_ftp.tsv:md5,94378c448c044b3e20e5c54e442ab62e",
            "SRR14593545.runinfo_ftp.tsv:md5,302271a4a1d99865362d0228e6439580",
            "SRR14709033.runinfo_ftp.tsv:md5,e059c4a0dfdddd70c1998a1ea9c1e65f",
            "SRR9984183.runinfo_ftp.tsv:md5,1e299191d23e0f3bd0aabf98c9d8133e"
        ],
        "timestamp": "2023-05-09T14:11:58+0000"
    },
    "viralrecon_samplesheet": {
        "content": [
            "id_mappings.csv:md5,40c6daa8d668b4930bfd9045248e16ec",
            "multiqc_config.yml:md5,a6370add94bbe60c6bbe1bb48a038e8d"
        ],
        "timestamp": "2023-05-09T14:11:58+0000"
    },
    "atacseq_metadata": {
        "content": [
            "DRR026872.runinfo_ftp.tsv:md5,c5ab92f6f50717b8163580b642b37aec",
            "DRR028935.runinfo_ftp.tsv:md5,9db5d6cba5ce2cccd9b2cc3faaa092e2",
            "ERR1109373.runinfo_ftp.tsv:md5,d1717cb1638950b7af22ec1b976616a4",
            "ERR1160846.runinfo_ftp.tsv:md5,5611622d583e6bc6d45d8f934696ff1e",
            "SRR12848126.runinfo_ftp.tsv:md5,9563f91d4cc6aa58d2ed5fba2eb30baf",
            "SRR13191702.runinfo_ftp.tsv:md5,94378c448c044b3e20e5c54e442ab62e",
            "SRR14593545.runinfo_ftp.tsv:md5,302271a4a1d99865362d0228e6439580",
            "SRR14709033.runinfo_ftp.tsv:md5,e059c4a0dfdddd70c1998a1ea9c1e65f",
            "SRR9984183.runinfo_ftp.tsv:md5,1e299191d23e0f3bd0aabf98c9d8133e"
        ],
        "timestamp": "2023-05-09T14:11:58+0000"
    },
    "rnaseq_ena_metadata": {
        "content": [
            "DRR026872.runinfo_ftp.tsv:md5,74a5f7058f27a565ce59db481355ca54",
            "DRR028935.runinfo_ftp.tsv:md5,07a98fdcec207686a4ea2c9a3997edf3",
            "ERR1109373.runinfo_ftp.tsv:md5,868897310855c50b1d5fc77d1195a1f9",
            "ERR1160846.runinfo_ftp.tsv:md5,146d7f2ec778b32b7da7ad040e554114",
            "SRR12848126.runinfo_ftp.tsv:md5,cf8827612a76645824e83418d7d76cbf",
            "SRR13191702.runinfo_ftp.tsv:md5,2031ca9b7ab736dbde37d05b10cdb0ad",
            "SRR14593545.runinfo_ftp.tsv:md5,3a1ac14d71744224a61ab542f869e180",
            "SRR14709033.runinfo_ftp.tsv:md5,2385a9cc5e320e78b02743c45d1338f1",
            "SRR9984183.runinfo_ftp.tsv:md5,f1ed9fc572d2e0d145be7f547f6d1236"
        ],
        "timestamp": "2023-05-09T14:11:58+0000"
    },
    "rnaseq_samplesheet": {
        "content": [
            "id_mappings.csv:md5,40c6daa8d668b4930bfd9045248e16ec",
            "multiqc_config.yml:md5,a6370add94bbe60c6bbe1bb48a038e8d",
            "samplesheet.csv:md5,9fa43672fb5ffcf0dec423a4627dc028"
        ],
        "timestamp": "2023-05-09T14:11:58+0000"
    },
    "rnaseq_samplesheet_specific_ena": {
        "content": [
            "id_mappings.csv:md5,0efde324e1818c0827390a17449a6e8a",
            "multiqc_config.yml:md5,6af65b0ae40e1bba60a25361771624bd",
            "samplesheet.csv:md5,dc7371849e112d0e95646ef1c34339d0"
        ],
        "timestamp": "2023-05-09T14:11:58+0000"
    },
    "rnaseq_metadata": {
        "content": [
            "DRR026872.runinfo_ftp.tsv:md5,c5ab92f6f50717b8163580b642b37aec",
            "DRR028935.runinfo_ftp.tsv:md5,9db5d6cba5ce2cccd9b2cc3faaa092e2",
            "ERR1109373.runinfo_ftp.tsv:md5,d1717cb1638950b7af22ec1b976616a4",
            "ERR1160846.runinfo_ftp.tsv:md5,5611622d583e6bc6d45d8f934696ff1e",
            "SRR12848126.runinfo_ftp.tsv:md5,9563f91d4cc6aa58d2ed5fba2eb30baf",
            "SRR13191702.runinfo_ftp.tsv:md5,94378c448c044b3e20e5c54e442ab62e",
            "SRR14593545.runinfo_ftp.tsv:md5,302271a4a1d99865362d0228e6439580",
            "SRR14709033.runinfo_ftp.tsv:md5,e059c4a0dfdddd70c1998a1ea9c1e65f",
            "SRR9984183.runinfo_ftp.tsv:md5,1e299191d23e0f3bd0aabf98c9d8133e"
        ],
        "timestamp": "2023-05-09T14:11:58+0000"
    },
    "rnaseq_specific_ena_samplesheet": {
        "content": [
            "id_mappings.csv:md5,0efde324e1818c0827390a17449a6e8a",
            "multiqc_config.yml:md5,6af65b0ae40e1bba60a25361771624bd",
            "samplesheet.csv:md5,6a08c28af1ed2629e18f0014f4d72ed6"
        ],
        "timestamp": "2023-05-09T14:11:58+0000"
    },
    "atacseq_samplesheet": {
        "content": [
            "id_mappings.csv:md5,40c6daa8d668b4930bfd9045248e16ec",
            "multiqc_config.yml:md5,a6370add94bbe60c6bbe1bb48a038e8d",
            "samplesheet.csv:md5,236f64ee8d40b3b658aecc0298cccfb2"
        ],
        "timestamp": "2023-05-09T14:11:58+0000"
    },
    "taxprofiler_samplesheet": {
        "content": [
            "id_mappings.csv:md5,40c6daa8d668b4930bfd9045248e16ec",
            "multiqc_config.yml:md5,a6370add94bbe60c6bbe1bb48a038e8d",
            "samplesheet.csv:md5,336e53f72cc538a58911880ec7c9b5da"
        ],
        "timestamp": "2023-05-09T14:11:58+0000"
    },
    "fastq": {
        "content": [
            "DRX024467_DRR026872.fastq.gz:md5,515e5448c55e054113c4a22df5f62791",
            "DRX026011_DRR028935_1.fastq.gz:md5,cc01df82a9354bb6b3be93483b20c35a",
            "DRX026011_DRR028935_2.fastq.gz:md5,941c9998d746416dad53c94c480ddf30",
            "ERX1188904_ERR1109373_1.fastq.gz:md5,9fd57225d6c07a31843276d6df9b15c0",
            "ERX1188904_ERR1109373_2.fastq.gz:md5,5a62e8f785687dce890cfb4fe3e607f9",
            "ERX1234253_ERR1160846.fastq.gz:md5,5924f20ef547ebdfed7cad795bbab6e6",
            "SRX10940790_SRR14593545_1.fastq.gz:md5,6a4762cf0f1f9f5544d3735e2e07c5d3",
            "SRX10940790_SRR14593545_2.fastq.gz:md5,cbd30715fdef457d7aa24a12f19eedf2",
            "SRX11047067_SRR14709033.fastq.gz:md5,79140dd1cb3fd1c5e02d88e01f7c3eda",
            "SRX6725035_SRR9984183.fastq.gz:md5,0b512d2dc31685983456bd56fd836544",
            "SRX9315476_SRR12848126_1.fastq.gz:md5,03f5a34586aef9cd61f54eb26831a665",
            "SRX9315476_SRR12848126_2.fastq.gz:md5,97701c5d884af8d96ba95d7237f3bf79",
            "SRX9626017_SRR13191702_1.fastq.gz:md5,89c5be920021a035084d8aeb74f32df7",
            "SRX9626017_SRR13191702_2.fastq.gz:md5,56271be38a80db78ef3bdfc5d9909b98",
            [
                "DRX024467_DRR026872.fastq.gz.md5:md5,dacaeba3498fb8f405411a22cdb4b40d",
                "DRX026011_DRR028935_1.fastq.gz.md5:md5,a47018c5972ba4c11539d0d6bc25eda9",
                "DRX026011_DRR028935_2.fastq.gz.md5:md5,f477bbef548fbed665734ebf084eea2f",
                "ERX1188904_ERR1109373_1.fastq.gz.md5:md5,50c1c3108631dc32a3151b67e0260d86",
                "ERX1188904_ERR1109373_2.fastq.gz.md5:md5,1f22463d410e91a288c83d43654b2499",
                "ERX1234253_ERR1160846.fastq.gz.md5:md5,513a8437ad331fc512b088b3750aa1ff",
                "SRX6725035_SRR9984183.fastq.gz.md5:md5,84f59dbb76c9b5eab7e14d90ced352c4",
                "SRX9626017_SRR13191702_1.fastq.gz.md5:md5,055a6916ec9ee478e453d50651f87997",
                "SRX9626017_SRR13191702_2.fastq.gz.md5:md5,c30ac785f8d80ec563fabf604d8bf945"
            ]
        ],
        "timestamp": "2023-05-09T14:11:58+0000"
    }
}